SpatialGDK/Source/Private/Utils/ComponentFactory.cpp

```diff
@@ -413,4 +430,70 @@ Worker_ComponentUpdate ComponentFactory::CreateHandoverComponentUpdate(Worker_Co
 	return ComponentUpdate;
 }
 
+Worker_ComponentData ComponentFactory::CreateInterestComponentData(UObject* Object, FClassInfo* Info)
+{
+	return CreateInterestComponent(Object, Info).CreateInterestData();
+}
+
+Worker_ComponentUpdate ComponentFactory::CreateInterestComponentUpdate(UObject* Object, FClassInfo* Info)
+{
+	return CreateInterestComponent(Object, Info).CreateInterestUpdate();
+}
+
+improbable::Interest ComponentFactory::CreateInterestComponent(UObject* Object, FClassInfo* Info)
+{
+	//Create a new component interest containing a query for every interested Object
+	improbable::ComponentInterest ComponentInterest;
+
+	for (FInterestPropertyInfo& PropertyInfo : Info->InterestProperties)
+	{
+		uint8* Data = (uint8*)Object + PropertyInfo.Offset;
+		if (UObjectPropertyBase* ObjectProperty = Cast<UObjectPropertyBase>(PropertyInfo.Property))
+		{
+			AddObjectToComponentInterest(Object, ObjectProperty, Data, ComponentInterest);
+		}
+		else if (UArrayProperty* ArrayProperty = Cast<UArrayProperty>(PropertyInfo.Property))
+		{
+			FScriptArrayHelper ArrayHelper(ArrayProperty, Data);
+			for (int i = 0; i < ArrayHelper.Num(); i++)
+			{
+				AddObjectToComponentInterest(Object, Cast<UObjectPropertyBase>(ArrayProperty->Inner), ArrayHelper.GetRawPtr(i), ComponentInterest);
+			}
+		}
+		else
+		{
+			checkNoEntry();
+		}
+	}
+
+	improbable::Interest Interest;
+	Interest.ComponentInterest.Add(improbable::Position::ComponentId, ComponentInterest);
+
+	return Interest;
+}
+
+void ComponentFactory::AddObjectToComponentInterest(UObject* Object, UObjectPropertyBase* Property, uint8* Data, improbable::ComponentInterest& ComponentInterest)
+{
+	UObject* ObjectOfInterest = Property->GetObjectPropertyValue(Data);
+
+	if (ObjectOfInterest == nullptr)
+	{
+		return;
```

Comment:
When would this happen?

Reply:
When a property goes from pointing to an object to being explicitly set to nullptr.